src/Lykke.Service.Decred.Api.Repository/AzureRepo.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AzureStorage;
+using Lykke.Service.Decred.Api.Common;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Lykke.Service.Decred.Api.Repository
+{
+    public class AzureRepo<T> : INosqlRepo<T> where T : TableEntity, new()
+    {
+        private const int RecordNotFoundStatus = 404;
+        private const int DuplicateRecordStatus = 409;
+        private const int InvalidRequestStatus = 500;
+
+        private readonly INoSQLTableStorage<T> _azureRepo;
+        
+        public AzureRepo(INoSQLTableStorage<T> azureRepo)
+        {
+            _azureRepo = azureRepo;
+        }
+
+        public async Task<bool> ExistsAsync(string key)
+        {
+            var t = new T
+            {
+                RowKey = key,
+                PartitionKey = "ByRowKey"
+            };
+            
+            return await _azureRepo.RecordExistsAsync(t);
+        }
+        
+        public async Task<T> GetAsync(string key)
+        {
+            return await _azureRepo.GetDataAsync("ByRowKey", key);
+        }
+
+        public async Task<IEnumerable<T>> GetAsync(IEnumerable<string> keys)
+        {
+            return await _azureRepo.GetDataAsync("ByRowKey", keys);
+        }
+
+        public async Task InsertAsync(T value, bool upsert)
+        {
+            try
+            {
+                if (upsert)
+                    await _azureRepo.InsertOrReplaceAsync(value);
+                else
+                    await _azureRepo.InsertAsync(value);
+            }
+            catch (StorageException e) when (e.RequestInformation.HttpStatusCode == DuplicateRecordStatus)
+            {
+                throw new BusinessException(ErrorReason.DuplicateRecord, $"{typeof(T)} already being observed", e);
```

Comment:
Why the word "observed" presents I base generic repository, that can be used for anything not related to observation at all?